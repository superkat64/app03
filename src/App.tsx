import React, {ReactElement} from 'react';
import {Logo, Button} from '@customink/pigment-react';
import './styles/App.scss';
// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-ignore
// eslint-disable-next-line import/no-unresolved
import RemoteHeader from 'ci_modular_header/Header';
import testingLogo from './octopus.png';

const App: React.FC = (): ReactElement => {
  return (
    <div className="App">
      <RemoteHeader />
      <header className="App-header">
        <Logo
          variant="inky"
          height="100px"
          onClick={() => undefined}
          className="App-logo"
        />
      </header>
      <div>
        <Button
          onClick={() => {
            window.location.href = 'https://reactjs.org';
          }}
          variant="primary">
          Learn React
        </Button>
        <div>
          <img src={testingLogo} alt="React Testing Library" />
        </div>
        <Button
          onClick={() => {
            window.location.href =
              'https://testing-library.com/docs/react-testing-library/intro/';
          }}
          variant="primary">
          React Testing Library
        </Button>
      </div>
    </div>
  );
};

export default App;
