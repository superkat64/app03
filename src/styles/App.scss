body {
  background-color: var(--color-gray-light);
  margin: 0;
}
