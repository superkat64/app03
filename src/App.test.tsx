import React from 'react';
import {render} from '@testing-library/react';
import App from './App';

test('renders learn react link', () => {
  const {container, getByText} = render(<App />);

  expect(getByText(/Learn/i)).toBeInTheDocument();
  expect(container.firstChild).toMatchSnapshot();
});
