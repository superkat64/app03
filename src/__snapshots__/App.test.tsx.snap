// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<div
  class="App"
>
  <header
    class="App-header"
  >
    <logo
      class="App-logo"
      height="100px"
      variant="inky"
    />
  </header>
  <div>
    <button
      variant="primary"
    >
      Learn React
    </button>
    <div>
      <img
        alt="React Testing Library"
        src="octopus.png"
      />
    </div>
    <button
      variant="primary"
    >
      React Testing Library
    </button>
  </div>
</div>
`;
